### PR TITLE
Warnings & 10 map funcs

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 10 functions (+0.09%) on `us`.

✅ This PR fixes 34 warnings!

File: src/world/area_sam/sam_05/D187E0.c [hunk 1]
```diff
@@ -1,11 +1,58 @@
...
+        sp50[i] = (cosine(script->functionTemp[0] + (i * 25) + 90) + 1.0) * 56.0;
+        sp68[i] = enemy->varTable[3];
+
+        func_802DE780(npc->spriteInstanceID, 0, 0xC, i, (sp20[i] << 24) | (sp38[i] << 16) | (sp50[i] << 8) | sp68[i], 0, 255, 0);
+    }
+
+    if (enemy->varTable[3] == 255) {
+        s32 vt3 = enemy->varTable[3]; // todo required to match
+
+        npc->renderMode = 0xD;
```
Comment (nanaian) [inline src/world/area_sam/sam_05/D187E0.c:48]: Enum

File: src/world/area_sam/sam_09/D362F0.c [hunk 2]
```diff
@@ -15,16 +15,70 @@ ApiStatus func_80240118_D36408(Evt* script, s32 isInitialCall) {
...
+    evt_set_variable(script, GameByte(268), mapVar3 & 0xFF);
+    evt_set_variable(script, GameByte(269), (mapVar3 >> 8) & 0xFF);
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_80240264_D36554(Evt* script, s32 isInitialCall) {
+    s32 temp_a2;
+    s32 temp_s0;
+
+    temp_s0 = evt_get_variable(script, GameByte(264)) & 0xFF;
```
Comment (nanaian) [inline src/world/area_sam/sam_09/D362F0.c:50]: IIRC all GameBytes were named, can we use the named variant?

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/world/area_sam/sam_09/D362F0.c (hunk 2 above)
Comment (nanaian) [reply src/world/area_sam/sam_09/D362F0.c:50]: GameByte(264) GB_SAM09_ItemSocket1_LowerByte

Comment (ethteck) [reply src/world/area_sam/sam_09/D362F0.c:50]: ah okay

**Pull request merged**